allennlp/tests/data/token_indexers/character_token_indexer_test.py

```diff
@@ -61,3 +61,88 @@ def test_start_and_end_tokens(self):
+        vocab.add_token_to_namespace("s", namespace="characters")  # 10
+        vocab.add_token_to_namespace("a", namespace="characters")  # 11
+        vocab.add_token_to_namespace("w", namespace="characters")  # 12
+        vocab.add_token_to_namespace("o", namespace="characters")  # 13
+        vocab.add_token_to_namespace("m", namespace="characters")  # 14
+        vocab.add_token_to_namespace(".", namespace="characters")  # 15
+
+        indexer = TokenCharactersIndexer("characters")
+        indices = indexer.tokens_to_indices(tokens, vocab, "char")
+        assert indices == {"char": [[2, 3, 3, 4, 5, 6, 7, 8],
+                                    [9, 10],
+                                    [11, 12, 4, 10, 13, 14, 4],
+                                    [15]]}
+        key_padding_lengths = "num_token_characters"
+        value_padding_lengths = 0
+        for token in indices["char"]:
+            item = indexer.get_padding_lengths(token)
+            key, value = item.keys(), item.values()
+            assert set([key_padding_lengths]) == set(key)
+            value_padding_lengths = max(value_padding_lengths, max(value))
+        assert value_padding_lengths == 8
+        padded = indexer.pad_token_sequence(indices,
+                                            {"char": len(indices["char"])},
+                                            {key_padding_lengths: value_padding_lengths})
+        assert padded == {"char": [[2, 3, 3, 4, 5, 6, 7, 8],
+                                   [9, 10, 0, 0, 0, 0, 0, 0],
+                                   [11, 12, 4, 10, 13, 14, 4, 0],
+                                   [15, 0, 0, 0, 0, 0, 0, 0]]}
+
+        indexer = TokenCharactersIndexer("characters", min_padding_length=3)
+        indices = indexer.tokens_to_indices(tokens, vocab, "char")
+        assert indices == {"char": [[2, 3, 3, 4, 5, 6, 7, 8],
+                                    [9, 10],
+                                    [11, 12, 4, 10, 13, 14, 4],
+                                    [15]]}
+        key_padding_lengths = "num_token_characters"
+        value_padding_lengths = 0
+        for token in indices["char"]:
+            item = indexer.get_padding_lengths(token)
+            key, value = item.keys(), item.values()
+            assert set([key_padding_lengths]) == set(key)
+            value_padding_lengths = max(value_padding_lengths, max(value))
+        assert value_padding_lengths == 8
+        padded = indexer.pad_token_sequence(indices,
+                                            {"char": len(indices["char"])},
+                                            {key_padding_lengths: value_padding_lengths})
+        assert padded == {"char": [[2, 3, 3, 4, 5, 6, 7, 8],
+                                   [9, 10, 0, 0, 0, 0, 0, 0],
+                                   [11, 12, 4, 10, 13, 14, 4, 0],
+                                   [15, 0, 0, 0, 0, 0, 0, 0]]}
```

Comment:
Lines 84 through 126 here test functionality that's already tested elsewhere.  The only thing we really need to test is the last case that you have, starting at line 128.